server/channels/app/platform/service.go

```diff
@@ -263,8 +267,26 @@ func New(sc ServiceConfig, options ...Option) (*PlatformService, error) {
 		}
 	}
 
+	ps.Store, err = ps.newStore()
+	if err != nil {
+		return nil, fmt.Errorf("cannot create store: %w", err)
+	}
+
+	// Step 7: Init License
+	if model.BuildEnterpriseReady == "true" {
+		ps.LoadLicense()
+	}
 	license := ps.License()
-	// Step 3: Initialize filestore
+
+	// This is a hack because ideally we wouldn't even have started the Redis client
+	// if the license didn't have clustering. But there's an intricate deadlock
+	// where license cannot be loaded before store, and store cannot be loaded before
+	// cache. So loading license before loading cache is an uphill battle.
+	if (license == nil || !*license.Features.Cluster) && *cacheConfig.CacheType == model.CacheTypeRedis && !ps.forceEnableRedis {
```

Comment:
Should we validate `forceEnableRedis` only if the service environment is either test or dev?

Reply:
I don't see too much value personally since the option is only enabled from the test code. And if you want, anybody can always comment out this check to enable it. But 0/5, let me know if you feel more strongly.

Reply:
Yeah, you're right, we can keep it as is.

Reply:
A note for future versions of ours, I'd recommend refactoring these bits (server initialization) like we did for the MPA into independent methods so that logic can resolve dependency tree on the fly and test deadlocks. It's actually a big broken window but this particular fix seems to be working now.

Reply:
Agree.